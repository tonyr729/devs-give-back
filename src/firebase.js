import firebase from 'firebase';

const config = {
  apiKey: "AIzaSyAVk4m35HpmPoi0Rr1K495lVyfskmVDiD4",
  authDomain: "devsgiveback.firebaseapp.com",
  databaseURL: "https://devsgiveback.firebaseio.com",
  projectId: "devsgiveback",
  storageBucket: "devsgiveback.appspot.com",
  messagingSenderId: "171321681886"
};
firebase.initializeApp(config);
export default firebase;