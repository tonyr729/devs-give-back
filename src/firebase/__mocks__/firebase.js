import * as firebase from 'firebase';

const onAuthStateChanged = jest.fn();

const getRedirectResult = jest.fn(() => {
  return Promise.resolve({
    user: {
      displayName: 'redirectResultTestDisplayName',
      email: 'redirectTest@test.com',
      emailVerified: true
    }
  })
})

const sendEmailVerification = jest.fn(() => {
  return Promise.resolve('result of sendEmailVerification');
})

const sendPasswordResetEmail = jest.fn(() => Promise.resolve());

const createUserWithEmailAndPassword = jest.fn(() => {
  return Promise.resolve('result of createUserWithEmailAndPassword');
})

const signInWithEmailAndPassword = jest.fn(() => {
  return Promise.resolve('result of signInWithEmailAndPassword');
})

const signInWithPopup = jest.fn(() => {
  return Promise.resolve('result of signInWithPopup');
})

const initializeApp = jest
  .spyOn(firebase, 'initializeApp')
  .mockImplementation(() => {
    return {
      auth: () => {
        return {
          createUserWithEmailAndPassword,
          signInWithEmailAndPassword,
          currentUser: {
            sendEmailVerification
          }
        }
      }
    }
  })

jest.spyOn(firebase, 'auth').mockImplementation(() => {
  return {
    onAuthStateChanged,
    currentUser: {
      displayName: 'testDisplayName',
      email: 'test@test.com',
      emailVerified: true
    },
    signInWithPopup
  };
});

firebase.auth.GithubAuthProvider = jest.fn(() => {})
firebase.auth.GoogleAuthProvider = jest.fn(() => {})

export default firebase;