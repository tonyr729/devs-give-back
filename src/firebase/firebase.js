import firebase from 'firebase';
import { firebaseKey } from './apiKeys';

const config = {
  apiKey: firebaseKey,
  authDomain: "devsgiveback.firebaseapp.com",
  databaseURL: "https://devsgiveback.firebaseio.com",
  projectId: "devsgiveback",
  storageBucket: "devsgiveback.appspot.com",
  messagingSenderId: "171321681886"
};
firebase.initializeApp(config);

export default firebase;