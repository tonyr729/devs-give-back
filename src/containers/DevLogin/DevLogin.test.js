import { shallow, mount } from 'enzyme';
import React from 'react';
import { 
  DevLogin, 
  mapStateToProps, 
  mapDispatchToProps 
} from './DevLogin';

import firebase from '../../firebase/firebase';
import { Redirect } from 'react-router-dom';
jest.mock('../../firebase/firebase');
jest.mock('../../helpers/DataCleaner.js', () => {
  return jest.fn().mockImplementation(() => {
    return {
      cleanDevLogin: jest.fn().mockImplementation(() => Promise.resolve({user: 'name'})),
      cleanError: jest.fn().mockImplementation(() => Promise.resolve({error: 'error'}))
    };
  });
});


describe('DevLogin', () => {
  it('should match snapshot', () => {
    const devLogin = shallow(<DevLogin dev={{}} />)

    expect(devLogin).toMatchSnapshot();
  });
  
  describe('gitHubLogin', () => {
    let mockProps;
    let devLogin;
    
    beforeEach(() => {
      mockProps = {
        dev: { },
        devError: jest.fn(),
        signInDev: jest.fn(),
        history: { push: jest.fn() }
      };
      
      devLogin = shallow(<DevLogin {...mockProps} />);
      devLogin.instance().database.gitHubLogin = jest.fn().mockImplementation(() => Promise.resolve({user: 'name'}));
    });
  
    it('should call the expected methods of firebase', () => {
      devLogin.instance().handleLogin()

      expect(devLogin.instance().database.gitHubLogin).toHaveBeenCalled();
    });
  });

  describe('logInCheck', () => {
    let mockProps;
    let devLogin;
    
    beforeEach(() => {
      mockProps = {
        dev: {user: 'tony'},
        devError: jest.fn(),
        signInDev: jest.fn(),
        history: { push: jest.fn() }
      };
      
      devLogin = shallow(<DevLogin {...mockProps} />);
      devLogin.instance().database.gitHubLogin = jest.fn().mockImplementation(() => Promise.resolve({user: 'name'}));
    });

    it('should call redirect if dev issigned in', () => {
      const expected = devLogin.instance().logInCheck(mockProps.dev)

      expect(expected).toEqual(<Redirect to='/dev-profile'/>);
    });
  });

  describe('handleLogin', () => {
    let mockProps;
    let devLogin;
    
    beforeEach(() => {
      mockProps = {
        dev: { },
        devError: jest.fn(),
        signInDev: jest.fn(),
        history: { push: jest.fn() }
      };
      
      devLogin = shallow(<DevLogin {...mockProps} />);
      devLogin.instance().database.gitHubLogin = jest.fn().mockImplementation(() => Promise.resolve({user: 'name'}));
    });
      
    
    it('should call the required functions when called', async () => {
      const instance = devLogin.instance()
      
      await instance.handleLogin()
      
      expect(instance.database.gitHubLogin).toHaveBeenCalled()
    });

    it('should call the required functions when there is an error', async () => {
      devLogin.instance().database.gitHubLogin = jest.fn().mockImplementation(() => Promise.reject({Error: 'error'}));
      const instance = devLogin.instance()
      
      await instance.handleLogin()
      
      expect(instance.cleaner.cleanError).toHaveBeenCalled();
      expect(instance.props.history.push).toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    it('returns an object with Dev info', () => {

      const mockState = {
        dev: {name: 'Tony'},
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockState);
      
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch on signInDevwith the correct params', () => {

      const mockDispatch = jest.fn();
      const user = {name: 'Tony'}
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'SIGN_IN_DEV',
        user
      };
      
      mappedProps.signInDev(user);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

    it('should call dispatch on DevError with the correct params', () => {

      const mockDispatch = jest.fn();
      const error = {error: 'error'}
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'DEV_ERROR',
        error
      };
      
      mappedProps.devError(error);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
      
    });
  });
});
