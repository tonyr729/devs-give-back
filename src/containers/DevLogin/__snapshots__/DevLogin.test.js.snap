// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevLogin should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DevLogin
    dev={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <p
          className="login-message"
>
          Please login
</p>,
        <button
          className="login-button-github"
          onClick={[Function]}
>
          Login with GitHub
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please login",
          "className": "login-message",
        },
        "ref": null,
        "rendered": "Please login",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login with GitHub",
          "className": "login-button-github",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Login with GitHub",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <p
            className="login-message"
>
            Please login
</p>,
          <button
            className="login-button-github"
            onClick={[Function]}
>
            Login with GitHub
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please login",
            "className": "login-message",
          },
          "ref": null,
          "rendered": "Please login",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login with GitHub",
            "className": "login-button-github",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Login with GitHub",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
